modules/travel_pay/spec/services/expenses_client_spec.rb

```diff
@@ -194,4 +179,111 @@
+      request_double = double('request', headers: {})
+
+      expect(connection_double).to receive(:delete)
+        .with("api/v1/expenses/other/#{expense_id}")
+        .and_yield(request_double)
+        .and_return(mock_response)
+
+      response = client.delete_expense(veis_token, btsss_token, expense_id, 'other')
+      expect(response.body['success']).to be(true)
+      expect(response.body['data']['id']).to eq(expense_id)
+    end
+
+    it 'raises an error when expense_id is not a valid UUID' do
+      expect { client.delete_expense(veis_token, btsss_token, 'not-a-uuid', 'other') }
+        .to raise_error(ArgumentError, /Invalid expense_id/)
+    end
+
+    it 'raises an error when expense type is unsupported' do
+      expect { client.delete_expense(veis_token, btsss_token, expense_id, 'unknown_type') }
+        .to raise_error(ArgumentError, /Unsupported expense type: unknown_type/)
+    end
+
+    it 'sets the correct headers' do
+      headers_hash = {}
+      request_double = double(headers: headers_hash)
+      mock_response = instance_double(Faraday::Response, status: 200, body: {})
+
+      allow(connection_double).to receive(:delete)
+        .with("api/v1/expenses/other/#{expense_id}")
+        .and_yield(request_double)
+        .and_return(mock_response)
+
+      client.delete_expense(veis_token, btsss_token, expense_id, 'other')
+
+      expect(headers_hash['Authorization']).to eq("Bearer #{veis_token}")
+      expect(headers_hash['BTSSS-Access-Token']).to eq(btsss_token)
+      expect(headers_hash['X-Correlation-ID']).to be_present
+    end
+
+    context 'when the API responds with errors' do
+      it 'handles a 400 Bad Request response' do
+        mock_response = instance_double(Faraday::Response, status: 400, body: { 'error' => 'Bad Request' })
+        allow(connection_double).to receive(:delete).and_return(mock_response)
+
+        response = client.delete_expense(veis_token, btsss_token, expense_id, 'other')
+        expect(response.status).to eq(400)
+        expect(response.body['error']).to eq('Bad Request')
+      end
+
+      it 'handles a 403 Forbidden response' do
+        mock_response = instance_double(Faraday::Response, status: 403, body: { 'error' => 'Forbidden' })
+        allow(connection_double).to receive(:delete).and_return(mock_response)
+
+        response = client.delete_expense(veis_token, btsss_token, expense_id, 'other')
+        expect(response.status).to eq(403)
+        expect(response.body['error']).to eq('Forbidden')
+      end
+
+      it 'handles a 404 Not Found response' do
+        mock_response = instance_double(Faraday::Response, status: 404, body: { 'error' => 'Not Found' })
+        allow(connection_double).to receive(:delete).and_return(mock_response)
+
+        response = client.delete_expense(veis_token, btsss_token, expense_id, 'other')
+        expect(response.status).to eq(404)
+        expect(response.body['error']).to eq('Not Found')
+      end
```

Comment:
So one thing to note: We have the Faraday middleware set up to raise on non-success responses. I wonder if these failure case tests would make sense to `expect(...).to raise(...)` instead.